crypto/objects/obj_dat.c

```diff
@@ -784,7 +787,23 @@ int OBJ_create(const char *oid, const char *sn, const char *ln)
         goto err;
     }
 
+    /*
+     * need to unlock so that OBJ_new_nid can take
+     * the same lock
+     */
+    ossl_obj_unlock(1);
```

Comment:
Rather than unlocking and relocking, could we instead create an internal "lockless" version of `OBJ_new_nid()` that could be called here. The public API `OBJ_new_nid()` would aquire the lock and call the lockless version.